### ISSUE TITLE
A11y problems w/ Carousel Pagination on ExploreData

### ISSUE THREAD

**Issue by benhammondmusic** [SatcherInstitute/health-equity-tracker#1040]
These issues were filed on the a117 report spreadsheet

- [x] Explore the data	SR	[A11Y - Health Equity Tracker]: No label associated with pagination buttons under Explore the Data page_Web_1.1	"ENVIRONMENT:
macOS 11.3 | Chrome Version 90.0.4430.212 (Official Build) (x86_64) | Voiceover
Chrome OS Version 91.0.4472.60 (Official Build) beta (64-bit) | Chromevox
PRE REQUISITES:
Turn on the respective screen reader
STEPS TO REPRODUCE:
  1). Open https://healthequitytracker.org/exploredata
  2). Linear navigate to the pagination buttons on the page
  3). Observe the announcement.
EXPECTED RESULT:
Buttons should have a label associated with it and the same should be assigned properly by screen reader.
OBSERVED RESULT:
Pagination buttons under Explore the Data page don't have label. Screen reader announces them only as button.
Replication video attached"	https://drive.google.com/file/d/1gJ61P0fK5CH1Npxp7aAk2NtQroOlUMBD/view?usp=sharing	"Issue is replicable in the following Android environment:
Device: Pixel 4
Android Version: 11
Chrome Browser Version: 91.0.4472.101"

- [x] Explore Data > Compare rates of x between y and z	SR	Missing label for page navigation dots.	"ENVIRONMENT:
OS | Browser | Screen Reader
macOS Big Sur:11.4|Version 90.0.4430.212 (Official Build) (x86_64) | VoiceOver
ChromeBook | Chrome Version: 90.0.4430.218 (Official Build) (64-bit) | ChromeVox
Window | Chrome Version 91.0.4472.77 (Official Build) (64-bit) | JAWS/NVDA
Window | Firefox Version: 87.0 (64-bit) | JAWS
PREREQUISITE:
Enable the respective screen reader.
STEPS TO REPRODUCE:
  1) Open this Url: https://healthequitytracker.org/ 
  2) Navigate to the explore the data tab and select.
  3) Navigate to the page navigation dots to observe this issue.
ACTUAL RESULT: Missing label for page navigation dots. Screen reader is announcing button only.
EXPECTED RESULT: Provide proper label for page navigation dots."	https://drive.google.com/file/d/1rZthVjjWhVo2m5-zApBoka9QgUflRLb0/view?usp=sharing

- [x] Explore Data > Compare rates of x between y and z	SR	Missing description of page navigation dots.	"ENVIRONMENT:
OS | Browser | Screen Reader
macOS Big Sur:11.4|Version 90.0.4430.212 (Official Build) (x86_64) | VoiceOver
ChromeBook | Chrome Version: 90.0.4430.218 (Official Build) (64-bit) | ChromeVox
Window | Chrome Version 91.0.4472.77 (Official Build) (64-bit) | JAWS/NVDA
Window | Firefox Version: 87.0 (64-bit) | JAWS
PREREQUISITE:
Enable the respective screen reader.
STEPS TO REPRODUCE:  
  1) Open this Url: https://healthequitytracker.org/ 
  2) Navigate to the explore the data tab and select.
  3) Navigate to the page navigation dots to observe this issue.
ACTUAL RESULT: Missing description about the page navigation dots. Screen reader is only announcing button but no indication that which page(dot) is selected.
EXPECTED RESULT: Provide an appropriate description to indicate the presence of dots and which page is selected."	https://drive.google.com/file/d/1rZthVjjWhVo2m5-zApBoka9QgUflRLb0/view?usp=sharing

- [x] Explore the data	DOM	[A11Y - Health Equity Tracker]: Improper DOM order under Explore the Data page_Web_1.8	"ENVIRONMENT:
macOS 11.3 | Chrome Version 90.0.4430.212 (Official Build) (x86_64) | Voiceover
Chrome OS Version 91.0.4472.60 (Official Build) beta (64-bit) | Chromevox
PRE REQUISITES:
Turn on the respective screen reader
STEPS TO REPRODUCE:
1). Open https://healthequitytracker.org/exploredata
2). Navigate through next & previous buttons on the top portion of the page
3). Observe the navigation & DOM order
EXPECTED RESULT:
Screen reader focus should move in a logical manner, ideally from left to right.
OBSERVED RESULT:
Screen reader focus goes to Next button and then to previous button, ideally it should move from previous to next button as previous button is placed before next button. 
Replication video attached"	https://drive.google.com/file/d/1t-YsMBv1VGFayBApF5_kYzKWI4eQV035/view?usp=sharing	"Issue is replicable in the following Android environment:
Device: Pixel 4
Android Version: 11
Chrome Browser Version: 91.0.4472.101"

- [x] Explore relationships between x and y in z	Keyboard	[A11Y - Health Equity Tracker]: Faint visual indicator for items in Explore relationship page when user navigates using tab navigation_Web_1.2	"ENVIRONMENT:
Platform: macOS 11.3
Browser: Chrome Version 90.0.4430.212 (Official Build) (x86_64)
STEPS TO REPRODUCE:
1). Open https://healthequitytracker.org/exploredata
2). Click on next button to reach the final sub page
3). Verify the options are selected as such, the page is 'Explore relationships between Diabetes and Covid-19 in United States.
4). Navigate through Explore Relationships page using tab key
5). Observe the visual indicator on the page. 
EXPECTED RESULT:
Element having keyboard focus should have a visual indicator demarcating the same.
OBSERVED RESULT:
Some elements on Explore Relationships page have a very faint visual indicator when user navigates using tab key. It is very difficult for a user to see the visual indicator and hence the keyboard focus for those elements.
Replication video attached"	https://drive.google.com/file/d/1CqU1wdASsr4mfUj1BCYRqqwjaXDdfksu/view?usp=sharing	Same issue is observed on Compare rates page.

- [x] US10	Touch Target	N/A	"Explore the data" Carousel button touch targets too small_mWeb(iOS)_2.1	"Environment:
DEVICE: iPad Air 2
iOS Version: 14.5.1
Model Number: MH0W2LL/A
Chrome browser: 87.0.4280.163
PREREQUISITE:
Tethered to MacOS with xcode & accessibility Scanner
STEPS TO REPRODUCE
1. Navigate to https://healthequitytracker.org/
2. Navigate to and activate the “Explore the Data” link link
3. Navigate to the ""next"" button and navigate to the last page of the dialogue
4. Navigate to the three dots (buttons) under the carousel
Actual Results: 
The touch target is too small. The Touch target is 16x16 but should be 44x44. The button is very hard to press in real application. The buttons are also very close together making the middle button art to touch. 
Expected Results:
Touch target should be 44x44 or at least large enough to touch when screen is zoomed.  "

- [x] 8	Screen reader		[a11y-HealthEquityTracker]Three buttons are missing accessibility labels_iPhone_1.1	"PREREQUISITE:
iPhone 12 Mini
iOS 14.5.1
Google Chrome 87.0.4280.163
1. Turn on VoiceOver screen reader
STEPS TO REPRODUCE
1. Navigate to https://healthequitytracker.org/
2. Navigate to and activate the “Expand Site Navigation” button
3. Navigate to and activate the “Explore the Data” link link
4. Observe that a dialogue launches
5. Navigate to the ""next"" button and navigate to the last page of the dialogue
6. Turn VoiceOver rotor to ""form controls""
7. Swipe down to navigate through the form controls found on the page
8. After the ""Previous"" button and the ""Share"" button, there are three additional buttons
9. Observe that VoiceOver reads them as buttons without accessibility labels                                                                                                                                                                                                                                                                                                            EXPECTED RESULT:
VoiceOver users use screen reader feedback to determine where they are on a page or application and interact with controls. Controls and images need meaningful and correctly associated accessibility labels, so that users can understand how to interact with these items. Accessibility labels are used by screen reader assistive technologies, such as Android TalkBack and iOS’ VoiceOver, and voice-activated assistive technology, such as Android’s Voice Access and iOS’ Voice Control. Based on the type of control that the developer uses to create elements in the UI, the assistive technology responds accordingly communicating the elements role, state, group info, and available user action to the user.
ACTUAL RESULT:
VoiceOver reports the form controls as  buttons without accessibility labels.
This repeats on multiple use cases including 8 and 9.

- [x] 9	Screen reader		[a11y-HealthEquityTracker]Buttons are labeled but their meaning is not obvious to the user_iPhone_1.5	"PREREQUISITE:
iPhone 12 Mini
iOS 14.5.1
Google Chrome 87.0.4280.163
1. Turn on VoiceOver screen reader
STEPS TO REPRODUCE
1. Navigate to https://healthequitytracker.org/
2. Navigate to and activate the “Expand Site Navigation” button
3. Navigate to and activate the “Explore the Data” link link
4. Observe that a dialogue launches
5. Navigate to the ""next"" button and navigate to the last page of the dialogue
6. Turn VoiceOver rotor to navigate by form controls
7. Swipe vertically three-four times to move from one button to the next
8. Observe that VoiceOver announces ""next"" and ""previous"" for buttons                                                                                                                                                                                                                                                                                             EXPECTED RESULT:
VoiceOver users use screen reader feedback to determine where they are on a page or application and interact with controls. Controls and images need meaningful and correctly associated accessibility labels, so that users can understand how to interact with these items. Accessibility labels are used by screen reader assistive technologies, such as Android TalkBack and iOS’ VoiceOver, and voice-activated assistive technology, such as Android’s Voice Access and iOS’ Voice Control. Based on the type of control that the developer uses to create elements in the UI, the assistive technology responds accordingly communicating the elements role, state, group info, and available user action to the user.
ACTUAL RESULT:
Even though there are accessibility labels on the two buttons, it is not immediately clear to the user what ""next"" and ""previous"" are supposed to do."

**Comment by benhammondmusic** [comment]
npm package providing the carousel component seems to not offer a solution. I subscribed to a filed issue and will research further: https://github.com/Learus/react-material-ui-carousel/issues/105

**Comment by benhammondmusic** [comment]
Order of "previous" and "next" buttons being rendered inside package... not sure it's possible nor necessary to reorder...
